Source/LuaBridge/RefCountedPtr.h

```diff
@@ -159,13 +159,17 @@ class RefCountedPtr : private detail::RefCountedPtrBase
     */
     void reset()
     {
-        if (m_p != 0)
+        auto myCount_it = getRefCounts().find(m_p); // use find() to avoid looking up m_p more than once
+        if (myCount_it != getRefCounts().end())
         {
-            if (--getRefCounts()[m_p] <= 0)
-                delete m_p;
-
-            m_p = 0;
+            if (--myCount_it->second <= 0)
+            {
+                if (m_p != 0)
+                    delete m_p;
+                getRefCounts().erase(m_p);
```

Comment:
You can increase the average performance by erasing the iterator instead of the value.
`getRefCounts().erase(myCount_it);`